Google Play Services crash on work profile

Am I just running too new version of gms?


```
type: crash
osVersion: google/husky/husky:14/AP2A.240805.005/2024080600:user/release-keys
package: com.google.android.gms:243233039
process: com.google.android.gms.persistent
processUptime: 3467399 + 338 ms
installer: com.android.vending
GmsCompatConfig version: 129

java.lang.SecurityException: getPackagesForUid: UID 1010142 requires android.permission.INTERACT_ACROSS_USERS_FULL or android.permission.INTERACT_ACROSS_USERS or android.permission.INTERACT_ACROSS_PROFILES to access user 0.
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3192)
	at android.os.Parcel.createException(Parcel.java:3176)
	at android.os.Parcel.readException(Parcel.java:3152)
	at android.os.Parcel.readException(Parcel.java:3094)
	at android.content.pm.IPackageManager$Stub$Proxy.getPackagesForUid(IPackageManager.java:5173)
	at android.app.ApplicationPackageManager$3.recompute(ApplicationPackageManager.java:1138)
	at android.app.ApplicationPackageManager$3.recompute(ApplicationPackageManager.java:1132)
	at android.app.PropertyInvalidatedCache.query(PropertyInvalidatedCache.java:999)
	at android.app.ApplicationPackageManager.getPackagesForUid(ApplicationPackageManager.java:1151)
	at com.android.internal.gmscompat.sysservice.GmcPackageManager.getPackagesForUid(GmcPackageManager.java:347)
	at bzsw.f(:com.google.android.gms@243233039@24.32.33 (190408-662265838):42)
	at bzsw.e(:com.google.android.gms@243233039@24.32.33 (190408-662265838):5)
	at bymi.run(:com.google.android.gms@243233039@24.32.33 (190408-662265838):997)
	at byme.run(:com.google.android.gms@243233039@24.32.33 (190408-662265838):64)
	at amse.c(:com.google.android.gms@243233039@24.32.33 (190408-662265838):50)
	at amse.run(:com.google.android.gms@243233039@24.32.33 (190408-662265838):76)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
	at amxq.run(:com.google.android.gms@243233039@24.32.33 (190408-662265838):8)
	at java.lang.Thread.run(Thread.java:1012)
	Suppressed: eoye: 
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.pm.ComputerEngine.enforceCrossUserOrProfilePermission(ComputerEngine.java:2826)
	at com.android.server.pm.IPackageManagerBase.getPackagesForUid(IPackageManagerBase.java:648)
	at android.content.pm.IPackageManager$Stub.onTransact(IPackageManager.java:2324)
	at com.android.server.pm.PackageManagerService$IPackageManagerImpl.onTransact(PackageManagerService.java:6554)
	at android.os.Binder.execTransactInternal(Binder.java:1534)
```

--- a/gmscompat_config
+++ b/gmscompat_config
@@ -210,6 +210,9 @@ unregisterTelephonyCallback void
 getServiceState null
 getVoiceMailNumber nullString
 
+[com.android.internal.gmscompat.sysservice.GmcPackageManager]
+getPackagesForUid nullArray
+
 [[versionMap]]
 # section name is gmscompat version, section contents are max supported versions of GmsCore and Play Store
 [1000]

--- a/src/app/grapheneos/gmscompat/GmsCompatConfigParser.java
+++ b/src/app/grapheneos/gmscompat/GmsCompatConfigParser.java
@@ -622,8 +622,8 @@ public class GmsCompatConfigParser {
                 stubCheck(signature, returnType.isArray()
                     && (
                         Parcelable.class.isAssignableFrom(returnType.getComponentType())
-                        ||
-                        returnType.getComponentType().isPrimitive()
+                        || returnType.getComponentType() == String.class
+                        || returnType.getComponentType().isPrimitive()
                     )
                 );
                 break;

